pipelines/candlepin/03-tests.yml

```
@@ -0,0 +1,17 @@
+---
+- name: run tests
+  become: True
+  hosts:
+   - "{{ forklift_name }}"
+  vars_files:
+    - ../vars/forklift_candlepin.yml
+  tasks:
+    - name: Run acceptance tests
+      ansible.builtin.include_role:
+        name: beaker
+        tasks_from: test
+      vars:
+        beaker_puppet_module: "puppet-candlepin"
+        beaker_os: "{{ pipeline_beaker_os }}"
+        beaker_environment:
+          BEAKER_FACTER_CANDLEPIN_BASEURL: "https://stagingyum.theforeman.org/candlepin/{{ pipeline_version }}/el{{ pipeline_beaker_os_version }}/x86_64"
```

technically, given this runs *on* the vagrant box, this could be
```suggestion
          BEAKER_FACTER_CANDLEPIN_BASEURL: "https://stagingyum.theforeman.org/candlepin/{{ pipeline_version }}/el{{ ansible_distribution_major_version }}/x86_64"
```

---

That won't work for testing EL8 and EL9  when the host itself is only EL8. That is why I felt these additional variables had to be introduced.

---

Agreed. I was working under the assumption we're keeping the current mode where we spin up a VM that matches the target OS.

I think I'm missing some bigger piece how this should be executed by automation.

---

Fair, while not efficient, that is how this workflow works and we could keep that connection and assumption.